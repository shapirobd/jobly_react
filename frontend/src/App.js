import React from "react";
import Routes from "./Routes";
import NavBar from "./NavBar";
import "./App.css";
import UserContext from "./UserContext";
import useLocalStorage from "./useLocalStorage";

function App() {
	// const [user, setUser] = useState(JSON.parse(localStorage.getItem("user")));
	// const [token, setToken] = useState(JSON.parse(localStorage.getItem("token")));
	// const [isLoggedIn, setIsLoggedIn] = useState(false);
	// const [loggingOut, setLoggingOut] = useState(false);

	const {
		user,
		setUser,
		token,
		setToken,
		isLoggedIn,
		setIsLoggedIn,
		loggingOut,
		setLoggingOut,
		isUpdated,
		setIsUpdated,
	} = useLocalStorage();

	return (
		<div className="App">
			<UserContext.Provider
				value={{
					user,
					setUser,
					token,
					setToken,
					isLoggedIn,
					setIsLoggedIn,
					loggingOut,
					setLoggingOut,
					isUpdated,
					setIsUpdated,
				}}
			>
				<NavBar />
				<Routes />
			</UserContext.Provider>
		</div>
	);
}

export default App;
