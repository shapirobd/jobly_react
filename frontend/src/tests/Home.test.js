import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import App from "../App";

test("smoke test", () => {
	render(
		<MemoryRouter initialEntries={["/"]}>
			<App />
		</MemoryRouter>
	);
});

test("snapshot test", () => {
	const { asFragment } = render(
		<MemoryRouter initialEntries={["/"]}>
			<App />
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});

test("contains correct content", () => {
	const { getByText } = render(
		<MemoryRouter initialEntries={["/"]}>
			<App />
		</MemoryRouter>
	);
	expect(
		getByText("All the jobs in one, convenient place.")
	).toBeInTheDocument();
	expect(getByText("Log in")).toBeInTheDocument();
});
