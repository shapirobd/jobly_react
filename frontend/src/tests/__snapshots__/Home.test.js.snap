// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="NavBar"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <button
            aria-label="menu"
            class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-5 MuiIconButton-colorInherit MuiIconButton-edgeStart"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <a
                class="makeStyles-title-6"
                href="/"
                variant="h6"
              >
                Jobly
              </a>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <div
            class="NavBar-routes"
          >
            <a
              href="/login"
            >
              Login
            </a>
            <a
              href="/signup"
            >
              Sign up
            </a>
          </div>
        </div>
      </header>
    </div>
    <div
      class="Home"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3"
      >
        Jobly
      </h3>
      <h5
        class="MuiTypography-root MuiTypography-h5"
      >
        All the jobs in one, convenient place.
      </h5>
      <a
        class="Home-btn"
        href="/login"
      >
        Log in
      </a>
      <a
        class="Home-btn"
        href="/signup"
      >
        Sign up
      </a>
    </div>
  </div>
</DocumentFragment>
`;
